src/plugins/gs-plugin-eos.c

```diff
@@ -354,8 +354,7 @@ gs_plugin_eos_blacklist_kapp_if_needed (GsPlugin *plugin, GsApp *app)
 	if (num_tokens < 2)
 		return FALSE;
 
-	/* since the last token should always be 'desktop' in Flatpak, we
-	 * need the one before that, which can be the locale */
+	/* last token may be the locale */
```

Comment:
@cosimoc my comment was not about how the index works. It's an outdated comment from a first version of the code where I was using the 'app-id' which has ends up always in '.desktop', so we'd take the token before the last one in that case.
I can fixup this commit if that's better.